### PR TITLE
feat(provider/kubernetes): support envFrom

### PR DESCRIPTION
**Pull request by ethanfrogers**
add support for envFrom pod option. this option allows users to specify
a config map of secret to use for environment variables.

image

@danielpeach PTAL!

spinnaker/spinnaker#2120

### CODE REVIEW

**Comment by ethanfrogers** [comment]
this is the corresponding clouddriver change: https://github.com/spinnaker/clouddriver/pull/2213

File: app/scripts/modules/kubernetes/src/container/environmentFrom.component.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    this.envFromSourceTypes.splice(index, 1);
+  }
+
+  public addEnvFrom(index: number): void {
+    this.envFrom.push({});
+    this.envFromSourceTypes.push(DEFAULT_ENVFROM_TYPE);
+  }
+
+  public updateEnvFrom(index: number): void {
+    let envFrom = this.envFrom[index];
```
Comment (danielpeach) [inline app/scripts/modules/kubernetes/src/container/environmentFrom.component.ts]: s/let/const here and below.

File: app/scripts/modules/kubernetes/src/container/environmentFrom.component.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      });
+  }
+
+  public removeEnvFrom(index: number): void {
+    this.envFrom.splice(index, 1);
+    this.envFromSourceTypes.splice(index, 1);
+  }
+
+  public addEnvFrom(index: number): void {
+    this.envFrom.push({});
```
Comment (danielpeach) [inline app/scripts/modules/kubernetes/src/container/environmentFrom.component.ts]: What you push here has to be of type `IKubernetesEnvironmentFrom`: `this.envFrom.push({ prefix: '' })`.

**Comment by ethanfrogers** [comment]
@danielpeach comments addressed!

**Pull request merged**